sycl/include/sycl/detail/accessor_iterator.hpp

```diff
@@ -8,12 +8,13 @@
 
 #pragma once
 
-#include <sycl/id.hpp>
+#include <sycl/access/access.hpp> // for mode, placeholder, target
+#include <sycl/buffer.hpp>        // for range
```

Comment:
Hm... should it be 
```suggestion
#include <sycl/range.hpp>         // for range
```
?
🤔